### PR TITLE
[Fluid] Compute Y+ Process

### PR DESCRIPTION
**Pull request by sunethwarna**
**📝 Description**
This PR Adds `ComputeYPlusProcess` and its python process output process named "YPlusOutputProcess" which can be used for the fly or post Y+ calculation. Appropriate tests are added to demonstrate the use case.

**🆕 Changelog**
- Added `Y_PLUS` variable to `FluidDynamicsApplication`
- Added `ComputeYPlusProcess` `c++` process
- Added `YPlusOutputProcess` python process
- Added tests

### CODE REVIEW

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (matekelemen) [inline applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:25]: We really need to add a proper container interface for ```PointerVectorSet```.

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (matekelemen) [inline applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:40]: how about something shorter, like ```"update_normals"```?

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (matekelemen) [inline applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:62]: minor thingy, but you can use [```enumerate``` with an offset](https://docs.python.org/3/library/functions.html#enumerate) in this case:
```py
        ref_y_values = {
            0.000255058,
            0.000277998,
            0.000262711,
            0.000262711,
            0.000277998,
            0.000255058,
            0.000300881,
            0.000300881,
            0.000331303,
            0.000331303,
        }

    for condition_id, y in enumerate(ref_y_values, 1):
```

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (matekelemen) [inline applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py]: again, ```enumerate```

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (matekelemen) [inline applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py]: ```process``` is going to be the last item from ```process_list``` from the iteration in line 111, but then it gets updated at the end of the iteration in line 122. I'm guessing ```process_list``` has only 1 item, but this is still a bit nasty.

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (matekelemen) [inline applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:129]: Minor again: I'd put deleting this file in ```setUp``` and ```tearDown``` to avoid complications when tests get interrupted (Ctrl+C, segfaults, etc.)

File: applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py
Comment (matekelemen) [inline applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py:40]: ```"output_step"``` is used as a frequency, so the name should indicate that

File: applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py
Comment (matekelemen) [inline applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py]: ```IntervalUtility``` can handle ```"End"``` as well
```
"interval" : [0.0, "End"]
```

File: applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py
Comment (matekelemen) [inline applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py:141]: We're gonna have to change ```TimeBasedAsciiFileWriterUtility``` to support appending files instead of keeping them open for the entire duration of the simulation at some point.

**Review by matekelemen — COMMENTED** [review]
Fine by me; minor comments only.

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (sunethwarna) [reply applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py]: nice catch. I did not notice it yet. I'll fix it.

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (sunethwarna) [reply applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:129]: makes sense. will do. 👍

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (sunethwarna) [reply applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:40]: I am ok with both. Historically we have been using `calculate_normals_every_time_step`. Lets see what others have to say.

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (sunethwarna) [reply applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:62]: I kept it like that to indicate that these are each conditions' y values. If we put enumerate, then it kind of gives the impression these will work only for consecutive condition ids.

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (sunethwarna) [reply applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py]: done :)

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (sunethwarna) [reply applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:129]: done :)

File: applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py
Comment (sunethwarna) [reply applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py:141]: good idea... may be for future 👍

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (matekelemen) [reply applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:62]: ah, allright

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (matekelemen) [reply applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:40]: I guess it shouldn't be changed if this name is used elsewhere too.

File: applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py
Comment (mpentek) [reply applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py:141]: > We're gonna have to change `TimeBasedAsciiFileWriterUtility` to support appending files instead of keeping them open for the entire duration of the simulation at some point.

For this actually I think of it as following: ASCII, VTK and GID output is good for fast postprocessing, even though perhaps not so efficient (either in size or having it open - general file handling). For everything else, there should be for example H5 output.

**Comment by mtnzguillermo** [comment]
I've just tested it and everything looks OK to me. :-)

File: applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py
Comment (mtnzguillermo) [reply applications/FluidDynamicsApplication/tests/test_compute_y_plus_process.py:40]: `"calculate_normals_every_time_step"` seems long but it is absolutely clear what it means. I'm not sure if I would understand it with `"update_normals"`. What about a slight improvement of the first one? Maybe `"update_normals_each_step"`.

**Comment by sunethwarna** [comment]
Can someone approve if everything is ok @rubenzorrilla @mpentek @matekelemen @mtnzguillermo ?

**Review by matekelemen — APPROVED** [review]
👍

**Comment by rubenzorrilla** [comment]
I've several concerns with this PR.

About the C++ compute Y-plus process, nothing to say. I think that it is a valuable contribution to the application. However, there are no unit tests of it. Note that it only tested through the outputting process, so if it eventually fails debugging would be more complicated. I suggest adding a simple cpp test for the process.

Concerning the outputting process, as far as I see, what it does is to compute the Y-plus to then output some statistics of it. If this is correct, the `FluidDynamicsApplication` is not the proper place for it, specially considering that you require the `StatisticsApplication` to use it (most of CFD users/developers use the `FluidDynamicsApplication` without the statistics one). 

Take into account that the `FluidDynamicsApplication` is a large (so hard to maintain) and key application, so the project requires to keep it as lightweight and controlled as possible.

As a comment aside I'd like to add that, we the @KratosMultiphysics/technical-committee decided to keep the codeowners for notification to make things more agile. However, it is customary to wait for the review of the application responsible(s) before merging, specially in the core applications. Note that we're in the summer vacations period, so reviews might take longer.

**Pull request merged**
